indico/modules/events/registration/controllers/management/fields.py

```diff
@@ -133,6 +136,14 @@ def _process(self):
         if (not enabled and self.field.type == RegistrationFormItemType.field_pd and
                 self.field.personal_data_type.is_required):
             raise BadRequest
+        if not enabled:
+            # Check this field is not conditional for other fields.
+            is_conditional = (db.session.query(RegistrationFormItem)
+                              .filter(RegistrationFormItem.data['show_if_field_id'].isnot(None))
+                              .filter(
+                cast(RegistrationFormItem.data['show_if_field_id'].astext, Integer) == self.field.id).has_rows())
```

Comment:
This wrapping looks ✨ wrong ✨ ;)

Reply:
I think this looks much nicer:

```py
            is_conditional = (
                RegistrationFormItem.query
                .filter(
                    RegistrationFormItem.data['show_if_field_id'].isnot(None),
                    cast(RegistrationFormItem.data['show_if_field_id'].astext, Integer) == self.field.id
                )
                .has_rows()
            )
```

Reply:
I hope it ✨ works ✨ better with [8eb06c9](https://github.com/indico/indico/pull/6678/commits/8eb06c950c13b1a6a1ce6069f7bc503cd9214a54).